From ab7e1451c48c2eab1b73eafa64922071e2c0d51a Mon Sep 17 00:00:00 2001
From: Jeff Mahoney <jeffm@suse.com>
Date: Wed, 18 Mar 2020 10:56:39 -0400
Subject: [PATCH 04/24] gdb: exclude PROP_UNDEFINED ranges from
 bound_flag_evaluated check

The bound_flag_evaluated flag will be set when a dynamic type is evaluated
as a static type.  If the high range is PROP_UNDEFINED, nothing will
be changed when it's evaluated, but the equality check will still fail.

This makes it impossible to compare types in Python code if the type
contains a zero-length array.

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 gdb/gdbtypes.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gdb/gdbtypes.c b/gdb/gdbtypes.c
index 3c29e7f4fb6..aea2d54e1a2 100644
--- a/gdb/gdbtypes.c
+++ b/gdb/gdbtypes.c
@@ -901,7 +901,8 @@ operator== (const range_bounds &l, const range_bounds &r)
   return (FIELD_EQ (low)
 	  && FIELD_EQ (high)
 	  && FIELD_EQ (flag_upper_bound_is_count)
-	  && FIELD_EQ (flag_bound_evaluated)
+	  && (FIELD_EQ (flag_bound_evaluated) ||
+	      l.high.kind == PROP_UNDEFINED)
 	  && FIELD_EQ (bias));
 
 #undef FIELD_EQ
-- 
2.26.2

